wrapper/src/main/java/software/amazon/jdbc/MonitorPluginService.java

```
@@ -0,0 +1,795 @@
+      hosts = hosts.stream()
+          .filter((hostSpec -> allowedHostIds.contains(hostSpec.getHostId())))
+          .collect(Collectors.toList());
+    }
+
+    if (!Utils.isNullOrEmpty(blockedHostIds)) {
+      hosts = hosts.stream()
+          .filter((hostSpec -> !blockedHostIds.contains(hostSpec.getHostId())))
+          .collect(Collectors.toList());
+    }
+
+    return hosts;
+  }
+
+  @Override
+  public void setAvailability(final @NonNull Set<String> hostAliases, final @NonNull HostAvailability availability) {
+
+    if (hostAliases.isEmpty()) {
+      return;
+    }
+
+    final List<HostSpec> hostsToChange = this.getAllHosts().stream()
+        .filter((host) -> hostAliases.contains(host.asAlias())
+            || host.getAliases().stream().anyMatch(hostAliases::contains))
+        .distinct()
+        .collect(Collectors.toList());
+
+    if (hostsToChange.isEmpty()) {
+      LOGGER.finest(() -> Messages.get("PluginServiceImpl.hostsChangelistEmpty"));
+      return;
+    }
+
+    final Map<String, EnumSet<NodeChangeOptions>> changes = new HashMap<>();
+    for (final HostSpec host : hostsToChange) {
+      final HostAvailability currentAvailability = host.getAvailability();
+      host.setAvailability(availability);
+      hostAvailabilityExpiringCache.put(host.getUrl(), availability,
+          DEFAULT_HOST_AVAILABILITY_CACHE_EXPIRE_NANO);
+      if (currentAvailability != availability) {
+        final EnumSet<NodeChangeOptions> hostChanges;
+        if (availability == HostAvailability.AVAILABLE) {
+          hostChanges = EnumSet.of(NodeChangeOptions.WENT_UP, NodeChangeOptions.NODE_CHANGED);
+        } else {
+          hostChanges = EnumSet.of(NodeChangeOptions.WENT_DOWN, NodeChangeOptions.NODE_CHANGED);
+        }
+        changes.put(host.getUrl(), hostChanges);
+      }
+    }
+
+    if (!changes.isEmpty()) {
+      this.pluginManager.notifyNodeListChanged(changes);
+    }
+  }
+
+  @Override
+  public boolean isInTransaction() {
```

I don't think that connections in monitoring threads will use transactions. We can simplify implementation here, like do nothing.

---

If we don't expect this to be used and don't want monitoring threads to call it I think we should throw an exception or log a warning and return (I guess return false?). For now I have the unexpected methods in this class throw exceptions so that I can see if they are being called in our integration tests, but we can decide whether we want them to log warnings instead.